utils/tracking.ts

```diff
@@ -0,0 +1,90 @@
+import { ServicesPreferencesModeEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/ServicesPreferencesMode";
+import { RetrievedProfile } from "@pagopa/io-functions-commons/dist/src/models/profile";
+import { EventTelemetry } from "applicationinsights/out/Declarations/Contracts";
+import { UpdateSubscriptionFeedInput } from "../UpsertServicePreferences/subscription_feed";
+import { initTelemetryClient } from "./appinsights";
+import { toHash } from "./crypto";
+
+export const createTracker = (
+  telemetryClient: ReturnType<typeof initTelemetryClient>
+) => {
+  const eventName = (name: string) => `api.profile.${name}`;
+
+  /**
+   * Trace an event when a user changes their preference mode
+   */
+  const traceServicePreferenceModeChange = (
+    hashedFiscalCode: string,
+    previousMode: ServicesPreferencesModeEnum,
+    nextMode: ServicesPreferencesModeEnum
+  ) =>
+    telemetryClient.trackEvent({
+      name: eventName("change-service-preferences-mode"),
+      properties: {
+        nextMode,
+        previousMode,
+        userId: hashedFiscalCode
+      },
+      tagOverrides: { samplingEnabled: "false" }
+    });
+
+  /**
+   * Trace an event when a user has previous preferences to migrate
+   */
+  const traceMigratingServicePreferences = (
+    oldProfile: RetrievedProfile,
+    newProfile: RetrievedProfile,
+    action: "REQUESTING" | "DOING" | "DONE"
```

Comment:
Could be useful add the serviceId only for `DOING` and `DONE` actions to check single preferences.

Reply:
Well, I think it does not. We may want to query users for which a migration started but it hasn't ended yet.

Otherwise, we could count the number of preferences to migrate at `REQUESTING` and then expecting the same number of `DONE` - but imho is too much for our needs

Reply:
You are right, I see now that only one time is called `DOING` and `DONE`, not one for service. If some preferences migration fail we can track logs errors events.